opentaxii/config.py

```diff
@@ -45,8 +74,13 @@ def __init__(self, optional_env_var=CONFIG_ENV_VAR, extra_configs=None):
         configs.append(self._get_env_config())
 
         options = self._load_configs(*configs)
-        if options['unauthorized_status'] not in ST_TYPES_10 + ST_TYPES_11:
-            raise ValueError('invalid value for unauthorized_status field')
+        options = self._clean_options(options)
+        if (
+            options["taxii1"]
+            and options["taxii1"]["unauthorized_status"]
+            not in ST_TYPES_10 + ST_TYPES_11
+        ):
+            raise ValueError("invalid value for unauthorized_status field")
```

Comment:
Because this package has a few `exceptions` modules, having single `Exception` subclass, `OpenTaxiiError` or something like that, and the rest of exception types subclass it in turn, would be a better design (unless it's a very local control flow based on `ValueError`). And instead of stdlib exceptions, at least `OpenTaxiiError` (or more specialised type when appropriate) is raised. Not sure, though, whether it's feasible in this PR.

Reply:
You make a good point. I checked the usage of exceptions and there's a lot of `ValueError` in there. I consider it a good suggestion to change those to be part of a logical structure, but it's a breaking change and out of scope for this PR so I'll create a new issue for it. Thanks for pointing it out!